conventions/src/main/kotlin/otel.java-conventions.gradle.kts

```diff
@@ -366,13 +366,6 @@ tasks.withType<Test>().configureEach {
   // This value is quite big because with lower values (3 mins) we were experiencing large number of false positives
   timeout.set(Duration.ofMinutes(15))
 
-  develocity.testRetry {
```

Comment:
I guess this will force us to deal with the flaky tests. Can always add it back if there are too many build failures.

Reply:
yeah, I was worried about us never seeing flaky tests if we don't remove it, but I realized a better option is probably to only disable retries on the daily builds in order to not disrupt other workflows (PRs / releases), will update this PR

Reply:
I updated this PR, can you give it one more look now? thanks